### PR TITLE
Borders and icons fix for Windows' high contrast mode

### LINKED ISSUE #408: [Global] Icons and borders are not visible in Windows High Contrast mode
## Issue summary

Control borders and UI icons are not visible in [Windows High Contrast mode](https://support.microsoft.com/en-us/windows/use-high-contrast-mode-in-windows-10-fedc744c-90ac-69df-aed5-c8a90125e696). This would cause difficulty in identifying pieces of UI, sizing of controls, and their purpose. Low-vision, high-contrast users should have equal access to content and imagery that sighted users have access to.

<details>
  <summary>Screenshots</summary>

Dawn header section, utility section is highlighted, icons are not visible.
Dawn product page, product accordion section is highlighted, icons are not visible.
Dawn footer section, email signup and social link sections are highlighted, icons are not visible.
Dawn login form, inputs are highlighted, borders are not visible.
Dawn search form, inputs are highlighted, borders and icon button is not visible.
Dawn cart page, quantity input and remove controls are highlighted, icons and borders are not visible.

</details>

<!-- ### Current code

#### HTML

```html

``` -->

## Steps to reproduce

1. Go to a Dawn theme store with any supported browser
2. With a Windows High Contrast theme activated, review the icons and button/input borders

## Behavior

### Expected

- Icons and control borders to be visible.

### Actual

- Icons are difficult to perceive or are not visible. Borders do not convey control sizing or affordance.

## Recommendation

1. In order for icons to be visible in High Contrast themes and not disrupt the intended design, use the [`forced-colors` media query](https://developer.mozilla.org/en-US/docs/Web/CSS/@media/forced-colors) to set the icon color based on the user preference.
2. To provide `button` and `input` sizing and appearance/affordance, apply a `transparent` border. This will allow a border to appear in High Contrast mode and not disrupt the intended design.

**Be sure to test all UI elements; the code snippet below may not cover all instances.**

### Recommended code

#### CSS

```css
@media screen and (forced-colors: active) {
  .icon {
    color: CanvasText;
    fill: CanvasText;
  }
}

.button, .field__input {
  border: solid 1px transparent;
}
```

<!-- #### Diff

```diff
- line removed
+ line added
``` -->

## Specifications

- **WCAG Principal:** Perceivable
- **WCAG SC:** [1.4.1 Use of Color](https://www.w3.org/WAI/WCAG21/Understanding/use-of-color.html)
- **Severity:** High
- **Effort:** Low

### PR DESCRIPTION
**Pull request by ludoboludo**
**Why are these changes introduced?**

Fixes #408 

**What approach did you take?**

Followed the suggestion from the issue. Using a media query specific to when high contrast mode is used. 

**Other considerations**

**Demo links**

- [Store](https://os2-demo.myshopify.com/?preview_theme_id=126382112790)
- [Editor](https://os2-demo.myshopify.com/admin/themes/126382112790/editor)

**Checklist**
- [ ] Followed [theme code principles](https://github.com/Shopify/dawn/blob/main/.github/CONTRIBUTING.md#theme-code-principles)
- [ ] Linted with [Theme Check](https://github.com/Shopify/theme-check)
- [ ] Tested on [mobile](https://shopify.dev/themes/store/requirements#mobile-browser-requirements)
- [ ] Tested on [multiple browsers](https://shopify.dev/themes/store/requirements#desktop-browser-requirements)
- [ ] Tested for [accessibility](https://shopify.dev/themes/best-practices/accessibility)

**Testing**
- Inputs, 
- Buttons,
- Icons (social media, header, collapsible tab), 
- Table (customer account), 
- loading spinner in cart

The loading spinner for the `add to cart` is not coming up properly but it's being changed in another PR to remove the disabled attribute (https://github.com/Shopify/dawn/pull/582).

### CODE REVIEW

**Comment by svinkle** [comment]
@ludoboludo Looking good. A few things:

1. Would it be possible to apply a `transparent` border for facet selectors?
2. Remove facet `x` icon is still difficult to see
3. Sort control border needs some padding
4. Facet checkbox is visually the same state; checked or unchecked (set `border` instead of `background-color` in media query)
   ```css
   @media screen and (forced-colors: active) {
     .facet-checkbox>svg {
       border: solid 1px rgb(var(--color-background));
     }
     .facet-checkbox>input[type=checkbox]:checked~.icon-checkmark {
       border: none;
     }
   }
   ```
5. Product selection is visually the same state; selected or unselected (add `transparent` underline)

<details>
  <summary>Screenshots</summary>
  
  
</details>

File: assets/template-collection.css [hunk 1]
```diff
@@ -185,6 +185,12 @@
   padding-right: 1.75rem;
 }
 
+@media screen and (forced-colors: active) {
+  .collection-filters__sort {
+    border: none;
+  }
+}
```
Comment (ludoboludo) [inline assets/template-collection.css:192]: @svinkle what about just removing the border overall ? So it has the same look as when it's not using windows' high contrast mode ?

**Review by svinkle — DISMISSED** [review]

File: assets/base.css [hunk 2]
```diff
@@ -1212,6 +1212,12 @@ deferred-media {
   color: transparent;
 }
 
+@media screen and (forced-colors: active) {
```
[the remark this answers is not in the document]
Comment (sofiamatulis) [reply assets/base.css:1215]: @ludoboludo The PR has been merged, so you will probably want to rebase it off of `main`

https://github.com/Shopify/dawn/pull/576

**Review by tauthomas01 — COMMENTED** [review]
Looks great! Thanks for doing this, I have never seen these new properties and media query so this is interesting 👍

File: assets/base.css (hunk 2 above)
[the remark this answers is not in the document]
Comment (ludoboludo) [reply assets/base.css:1243]: Done! Thanks for bringing it up 👌

**Review by tauthomas01 — APPROVED** [review]
LGTM

**Comment by ludoboludo** [comment]
Sorry @svinkle & @tauthomas01, I just pushed one last thing I noticed. When adding to the cart, while the submit button had `aria-disabled="true"`, it should be hiding the `Add to cart` text and showing only the loading spinner. But it was still showing the text under the spinner. With the [latest commit](https://github.com/Shopify/dawn/pull/619/commits/62be7cb6db118f44d00e8f2db02bc675c242bc51) it's fixed 👌

**Review by tauthomas01 — DISMISSED** [review]

**Review by svinkle — APPROVED** [review]

**Comment by ludoboludo** [comment]
Annnnd same thing again. Facet filter on search page got merged so it created some conflicts I had to fix. Sorry for the re request. 😬 
@svinkle @tauthomas01

**Comment by svinkle** [comment]
LGTM

**Review by tauthomas01 — APPROVED** [review]
Facets work well

**Pull request merged**